libverify/verify_libfuzzer.cpp

```diff
@@ -0,0 +1,1042 @@
+/**
+ * Defines an interactive implementation of assume for libfuzzer. When
+ * assumption fails, the reason is logged, and complete the requirements
+ * @date 2019
+ */
+
+#include "verify.h"
+#include "verify_fuzzer.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <stddef.h>
+#include <setjmp.h>
+#include <assert.h>
+
+
+void sol_assert(sol_raw_uint8_t _cond, const char* _msg)
+{
+    if( !_cond){
+		fprintf(stderr,"assert");
+		if(_msg) fprintf(stderr,":%s", _msg);
+		fprintf(stderr,"\n");    
+    }
+        assert(_cond);
+}
+
+void sol_require(sol_raw_uint8_t _cond, const char* _msg)
+{
+	if( !_cond){
+		fprintf(stderr,"require");
+		if(_msg) fprintf(stderr,":%s", _msg);
+		fprintf(stderr,"\n");
+    }
+		ll_assume(_cond);	
+}
+
+void ll_assume(sol_raw_uint8_t _cond){
+    if(!_cond){
+
+	/*exception_type 2: 
+	 * the ll_assume is failure*/
+	exception_type = 2;
+	TerminateExploration(exception_type);	
+    }
+}
+
+/*
+ * declare the global array RandData 
+ * with SizeOfRandData size, 
+ * global Counter CounterOfRandData */
+uint8_t *RandData;
+size_t SizeOfRandData = 0;
+size_t CounterOfRandData = 0;
+
+
+/*set up the exploration with Env environment,
+ * return the result of setjmp function*/
+int SetupExploration (void){
+	int r;
+	r = setjmp(Env);
+	return r;
+}
+
+/*terminate the exploration with Env environment 
+ * and different exception types, 
+ * which can jump to setjmp with different results*/
+void TerminateExploration (int exception_type){
+	longjmp(Env, exception_type);
+}
+
+/*
+ * whether the global array run out */
+void whether_RunOut(void)
```

Comment:
@huxinwen1994 I think it would be easier to define a function `tryGetNextRandByte` and use this everywhere you need more bytes (e.g., `nd_int8_t`, etc.):
```
uint8_t tryGetNextRandByte() {
    if (CounterOfRandData >= SizeOfRandData) {
        TerminateExploration(1);
    }
    uint8_t ret = RandData[CounterOfRandData];
	CounterOfRandData++;
    return ret;
}

Reply:
OK